src/ralph/data_center/views.py

```diff
@@ -0,0 +1,23 @@
+from ralph.admin.views.extra import RalphDetailView
+
+
+class RelationsView(RalphDetailView):
+    icon = 'shekel'
+    label = 'Relations'
+    name = 'relations'
+    url_name = 'relations'
+    template_name = 'data_center/datacenterasset/relations.html'
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        related_objects = {}
+
+        if self.object.cloudhost_set.exists():
```

Comment:
couldn't you just fetch all cloudhosts (even if there is none) - now you're making two queries, while you could do this in one?

Reply:
I should probably do that because I suspect exists() will actually rise an exception, if the DCA does not host any CH.

Reply:
if `.exists()` will raise an exception if there is non CH attached to DCA, you'll get 500 instead of `No related objects found.`, wouldn't you?

Reply:
yup, that's why I've changed the current approach in my next patch set